apptuit/apptuit_client.py

```diff
@@ -83,13 +105,42 @@ def __init__(self, token, api_endpoint="https://api.apptuit.ai/", debug=False):
             port: Port on which the service is running
 
         """
-        if not token:
-            raise ValueError("Invalid token")
         self.token = token
+        if not self.token:
+            self.token = _get_token_from_environment()
+            if not self.token:
+                raise ValueError("Missing Token, '" + APPTUIT_API_TOKEN + "'"
+                                 "is not available in environment variable or"
+                                 "should pass token as parameter to Apptuit")
         self.endpoint = api_endpoint
         if self.endpoint[-1] == '/':
             self.endpoint = self.endpoint[:-1]
         self.debug = debug
+        self._environ_tags = _get_tags_from_environment()
+
+    def _create_payload(self, datapoints):
+        data = []
+        for dp in datapoints:
+            if dp.tags and self._environ_tags:
+                tags = self._environ_tags.copy()
+                tags.update(dp.tags)
+            elif dp.tags:
+                tags = dp.tags
+            else:
+                tags = self._environ_tags.copy()
+            if not tags:
+                raise ValueError("Missing tags for:'"
+                                 + dp.metric +
+                                 "'. Pass parameter tags to DataPoint or set an Environment "
+                                 "Variable '"
+                                 + APPTUIT_PY_TAGS + "'.")
```

Comment:
The exception message could be :
"Missing tags for the metric " + dp.metric + ". Either pass it as value of the tags parameter to DataPoint or set environment variable APPTUIT_PY_TAGS for global tags"